builder/site.hs

```diff
@@ -1,15 +1,37 @@
 --------------------------------------------------------------------------------
+{-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE OverloadedStrings #-}
-import           Data.Monoid           ((<>))
+import           Data.Aeson
+import qualified Data.ByteString.Lazy          as BL
+import           Data.Monoid                   ((<>))
 import           Data.Time.Calendar
 import           Data.Time.Clock
 import           Hakyll
+import           Hakyll.Core.Compiler
+import           Hakyll.Core.Compiler.Internal
+import           Hakyll.Core.Provider
 import           System.FilePath.Posix
+import           Testimonial
 
 
 --------------------------------------------------------------------------------
 main :: IO ()
 main = mkContext >>= \ctx -> hakyll $ do
+  match "testimonials/logos/*" $ do
+    route idRoute
+    compile copyFileCompiler
+
+  match "testimonials/*.yaml" $ do
+    compile parseTestimonialCompiler
+
+  create ["testimonials.json"] $ do
+    route idRoute
+    compile $ do
+      testimonials <- loadAll @Testimonial "testimonials/*.yaml"
+      item <- (makeItem . BL.unpack . encode . map itemBody) testimonials
```

Comment:
I'm not following this code exactly: why do we need the `BL.unpack`? What are we using a `[Word8]` for?

Reply:
We load all of the testimonials, then serialize them to json and convert the json to raw bytes that we can save to disk. There are a bunch of other ways we might have done this. It could probably use a comment. Since we're intending to upgrade hakyll and refactor some of this code, I'll ensure we more thoroughly comment whatever the resulting refactored code is.